taichi/gui/x11.cpp

```diff
@@ -73,6 +74,14 @@ void GUI::process_event() {
     switch (ev.type) {
       case Expose:
         break;
+      case ClientMessage:
+        // https://stackoverflow.com/questions/10792361/how-do-i-gracefully-exit-an-x11-event-loop
+        if (ev.xclient.data.l[0] == *(Atom *)wmDeleteMessage.data()) {
+          key_events.push_back(
```

Comment:
Could you wrap this logic as a member function for `GUI`? I implemented something for Cocoa, and had to reuse this bit of logic..